source/appModules/soffice.py

```diff
@@ -433,6 +474,35 @@ def _backspaceScriptHelper(self, unit: str, gesture: inputCore.InputGesture):
 		else:
 			log.warning('Backspace did not remove text as expected.')
 
+	def script_toggleTextAttribute(self, gesture):
+		# reset time and enable announcement of toggled toolbar buttons
+		# s. SymphonyButton#event_stateChange
+		SymphonyDocument.announce_toolbar_button_toggle = True
+		SymphonyDocument.last_formatting_gesture_event_time = time.time()
+		# send gesture
+		gesture.send()
+
+	__gestures = {

```

Comment:
The preferred method to associate gestures to scripts is the `@script` decorator; could you use it? Or is there any reason why you did not?

Note that you can define the `@script` decorator with only the `gesture` or the `gestures` parameters, i.e. you can omit `description` and `category` so that the script is not remappable.

Reply:
Thanks, I've updated this accordingly.
(The only reason I didn't do that was that I previously looked at another example that didn't (`script_toggleBold` in `source/NVDAObjects/window/winword.py`) and did the same as that one, rather than starting from scratch using the [NVDA developer guide where this is explained](https://www.nvaccess.org/files/nvda/documentation/developerGuide.html#DefiningScriptProperties).)